packages/sources/refund.move

```diff
@@ -0,0 +1,514 @@
+    public fun current_liabilities(pool: &RefundPool): u64 { accounting::current_liabilities(&pool.accounting) }
+
+    // === Friends ===
+
+    public(friend) fun uid_mut(pool: &mut RefundPool): &mut UID { &mut pool.id }
+    public(friend) fun unclaimed_mut(pool: &mut RefundPool): &mut Table<address, u64> { &mut pool.unclaimed }
+    public(friend) fun accounting_mut(pool: &mut RefundPool): &mut Accounting { &mut pool.accounting }
+    public(friend) fun booster_pool_mut(pool: &mut RefundPool): &mut Pool { &mut pool.booster_pool }
+
+    public(friend) fun reclaim_funds_(
+        inner_pool: &mut Pool,
+        total_raised: u64,
+        total_unclaimed: u64,
+        ctx: &mut TxContext,
+    ) {
+        let funders = funders_mut(inner_pool);
+        assert!(table::contains(funders, sender(ctx)), EInvalidFunder);
+        let funding_amount = table::remove(funders, sender(ctx));
+
+        let is_last = table::is_empty(funders);
+        let funds = funds_mut(inner_pool);
+        
+        let reclaim_amount = if (is_last) {
+            balance::value(funds)
+        } else {
+            // ReclaimAmount = Leftovers * % Share <=>
+            // ReclaimAmount = Leftovers * FundingAmount/TotalRaised
+            // 
+            // We first upscale then downscale
+            mul_div(funding_amount, total_unclaimed, total_raised)
+
+        };
+
+        let reclaim_funds = coin::from_balance(balance::split(funds, reclaim_amount), ctx);
+        transfer::public_transfer(reclaim_funds, sender(ctx));
+    }
+
+    // === Phase Transitions ===
+    
+    public entry fun start_funding_phase(
+        pub: &Publisher,
+        pool: &mut RefundPool,
+        timeout_ts: u64,
+        clock: &Clock,
+    ) {
+        assert_publisher(pub);
+        assert_address_addition_phase(pool);
+        assert!(!table::is_empty(&pool.unclaimed), ERefundPoolHasZeroAddresses);
+        
+        assert!(timeout_ts >= clock::timestamp_ms(clock) + MIN_DEFAULT_PERIOD, EInvalidTimeoutTimestamp);
+        option::fill(&mut pool.timeout_ts, timeout_ts);
+
+        next_phase(pool)
+    }
+
+    public entry fun start_claim_phase(
+        pool: &mut RefundPool,
+    ) {
+        assert_funding_phase(pool);
+        let total_to_refund = total_to_refund(&pool.accounting);
+        let total_raised = total_raised(&pool.accounting);
+        assert!(total_to_refund == total_raised, EPoolUnderfunded);
+
+        next_phase(pool)
+    }
+    
+    public entry fun start_reclaim_phase(
+        pool: &mut RefundPool,
+        clock: &Clock,
+    ) {
+        let timeout_ts = option::borrow(&pool.timeout_ts);
+        assert!(clock::timestamp_ms(clock) >= *timeout_ts, ECurrentTimeBeforeTimeout);
+
+        if (is_funding_phase(pool)) {
+            let total_to_refund = total_to_refund(&pool.accounting);
+            let total_raised = total_raised(&pool.accounting);
+            assert!(total_raised < total_to_refund, 0);
+            
```

Comment:
**The same comment as on the other PR applies here.** 

We need to move from funding phase directly into reclaim phase, not first into "claim" phase, otherwise you could theoretically bump it to "claim", do a few claims and then move into reclaim.

Reply:
I believe this is related your comment on the timeout check for `reclaim_funds` endpoints, correct? If so, is has been addressed by the new implemented timeout check in these endpoints.

Reply:
Ok, I added an extra default time, for when setting the `timeout_ts` in the `start_funding_phase`:

`assert!(timeout_ts >= clock::timestamp_ms(clock) + MIN_FUNDING_PERIOD_MS + MIN_CLAIM_PERIOD_MS, EInvalidTimeoutTimestamp);`

That way we at least guarantee that the timeout_ts accounts for both periods. Both default periods are set to `259_200_000` corresponding to 3 days in milliseconds.

In the `start_claim_phase` I then added:

```
let current_ts = clock::timestamp_ms(clock);

// We check if there's at least 3 days difference between the clock and the timeout.
// If not, we bump the timeout_ts to correspond to a least the minimum period
if (timeout_ts - current_ts >= MIN_CLAIM_PERIOD_MS) {
    option::swap(&mut pool.timeout_ts, current_ts + MIN_CLAIM_PERIOD_MS);
};
```

This way the claim phase is guaranteed to have have the timespan of at least the minimum default time.